### ISSUE TITLE
TST: Update visualization test data for CircleCI mpldev

### ISSUE THREAD

**Issue by pllim** [astropy/astropy#11471]
Now that `matplotlib` dev bugs are fixed upstream, there are real failures in the CircleCI `mpldev` job that we need to fix, namely updating the data files, as they have bumped from 3.3.dev to 3.4.dev. You will see errors like this until this is fixed:

```
UserWarning: Downloading http://data.astropy.org/testing/astropy/2019-08-02T11:38:58.288466/3.4.x/....png
failed: HTTP Error 404: Not Found
```

The data are in https://github.com/astropy/astropy-data/tree/gh-pages/testing/astropy/2019-08-02T11:38:58.288466 but I am not familiar with the procedures to do this update myself.

* Do you create a new dir with new timestamp or a new subdir under this timestamp? When do you choose to or not to create new dirs with new timestamps?
* If copying the expected outputs from `3.3.x` to `3.4.x` is insufficient, how do we update the data files? I don't see the CircleCI job for `mpldev` uploading any artifacts into CircleCI itself in the failed job: `Uploading /root/project/results to results... No artifact files found at /root/project/results`
* Are we supposed to update the CircleCI `mpl` test matrix too? Currently, it only runs `302`, `310`, and `dev`. When do we choose to update matrix and how to we pick the versions that are deemed important enough to test in CI?

The `mpldev` job isn't as critical as other jobs but we should still fix this as soon as possible, because it is failing for unrelated PRs, causing unnecessary stress when PR authors see the ❌ .

**Comment by pllim** [comment]
Attempted fix at astropy/astropy-data#109 and astropy/astropy-data#110

**Comment by pllim** [comment]
✔️
